models/transaction.go

```diff
@@ -246,15 +246,15 @@ func decodeTransactionEvent(
 	if txEvent.TransactionType == types.DirectCallTxType {
 		directCall, err := types.DirectCallFromEncoded(encodedTx)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to RLP-decode direct call: %w", err)
+			return nil, nil, fmt.Errorf("failed to RLP-decode direct call [%x]: %w", encodedTx, err)
```

Comment:
It occurred to me that we have the same exact error message in `decodeTransactionEvent` & `UnmarshalTransaction`:

In `decodeTransactionEvent`:
```go
directCall, err := types.DirectCallFromEncoded(encodedTx)
if err != nil {
	return nil, nil, fmt.Errorf("failed to RLP-decode direct call [%x]: %w", encodedTx, err)
}

gethTx := &gethTypes.Transaction{}
if err := gethTx.UnmarshalBinary(encodedTx); err != nil {
	return nil, nil, fmt.Errorf("failed to RLP-decode transaction [%x]: %w", encodedTx, err)
}
```

In `UnmarshalTransaction`:
```go
directCall, err := types.DirectCallFromEncoded(value)
if err != nil {
	return nil, fmt.Errorf("failed to RLP-decode direct call [%x]: %w", value, err)
}

if err := tx.UnmarshalBinary(value); err == nil {
	return TransactionCall{Transaction: tx}, nil
}

return nil, fmt.Errorf("failed to RLP-decode transaction [%x]: %w", value, err)
```

Can we add a prefix to the error message, so we can know whether the decoding error comes the ingestion component, or from the DB?

Reply:
no need because the error in the API gets wrapped.